src/Utilities/Compiler/GeneratedCodeUtilities.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+
+namespace Analyzer.Utilities
+{
+    internal static class GeneratedCodeUtilities
+    {
+        private static readonly string[] s_autoGeneratedStrings = new[] { "<autogenerated", "<auto-generated" };
+
+        internal static bool IsGeneratedCode(SyntaxTree tree, Func<SyntaxTrivia, bool> isComment, CancellationToken cancellationToken)
+        {
+            return BeginsWithAutoGeneratedComment(tree, isComment, cancellationToken);
+        }
+
+        internal static bool IsNotGeneratedCodeOrRazorPage(SyntaxTree tree, Func<SyntaxTrivia, bool> isComment, CancellationToken cancellationToken)
```

Comment:
Only this method is used for now, could remove unused `IsGeneratedCode(...)` overloads

Reply:
Removed all unrelated code